feat:  extend tracking performance scripts

asalzburger opened the pull request:
This PR:
 * extends the functionality of `Examples/Scripts/TrackingPerformance/boundParamResiduals`,
 * creates a compiled standalone version as `ActsExampleResidualAndPulls`
 * adds `ACTS_BUILD_EXAMPLES_SCRIPTS` CMake flag

It also fixes the issue of `phi` residuals that did not respect periodic scaling.

asalzburger commented:
![phi_pull](https://user-images.githubusercontent.com/26623879/117309519-c6d8a380-ae82-11eb-8959-dd034b052c5c.png)

FYI: Phi-pulls before the fix.

XiaocongAi commented on Examples/Scripts/TrackingPerformance/boundParamResolution.C:
```diff
@@ -0,0 +1,852 @@
+      return {std::string("all_"), ""};
+    }
+
+    std::string vlstr = "vol" + std::to_string(vol);
```
It seems like we should have the volume id cut only if `vol > 0`, and have the layer id cut only if `vol > 0 and layer > 0`, e.g.  an input of `vol =  -1` and `lay = 8` can mean that no cut is necessary ?

asalzburger replied:
if `volume_id < 0` then necessarily `layer_id < 0`, hence this case is covered by

```c++
    if (vol < 0 and lay < 0) {
      return {std::string("all_"), ""};
    }
```

In L366.

XiaocongAi commented on Examples/Scripts/TrackingPerformance/boundParamResolution.C:
```diff
@@ -0,0 +1,852 @@
+    }
+  }
+
+  // Section: Histogram filling
```
```suggestion
  // Section 4: Histogram filling
```

asalzburger replied:
Fixed.

XiaocongAi commented on Examples/Scripts/TrackingPerformance/boundParamResolution.C:
```diff
@@ -0,0 +1,852 @@
+    }
+  }
+
+  // Section: Histogram plotting
```
```suggestion
  // Section 5: Histogram plotting
```

asalzburger replied:
Also fixed.

XiaocongAi reviewed (commented):
Hi @asalzburger , thank you for this really nice PR. I have only a few minor comments.

asalzburger commented on Examples/Scripts/TrackingPerformance/boundParamResolution.C:
```diff
@@ -0,0 +1,852 @@
+  TColor::CreateGradientColorTable(number, stops, red, green, blue, n);
+}
+
+// loc1, phi, theta, q/p, t) at all track states from root file produced by the
```
Added '('

asalzburger commented:
Hi @XiaocongAi  - I addressed the PR comments, please have another look if you're ok with that. I might extend the functionality of this script in the future, but I think that can go into another PR.

asalzburger commented:
I've resolved the merge conflict, will merge when CI runs through.
